### ISSUE TITLE
Whatsapp shortcuts not coming by default in redmi note 8 pro

### ISSUE THREAD

**Issue by dhawalmehta22** [BranchMetrics/sesame_issues#163]
![1583526837436](https://user-images.githubusercontent.com/43503404/76120651-f3432680-6017-11ea-9d89-ef9baf4f4c43.jpg)
This shortcut from whatsapp is not coming in redmi note 8 pro

**Comment by philwall2** [comment]
Let's figure this out.

1. Which Shortcut?
2. How are you accessing it? Are you using the Sesame Search Widget? Nova App Search?

Which shortcut?

**Comment by dhawalmehta22** [comment]
The screen which i shared is from another device. In which you can see all the contacts are by default created as shortcuts.
And now i am sharing my phones screenshot the screen which i am getting when i tap whatsapp in the shortcuts tab.
![Screenshot_2020-03-09-22-30-23-270_ninja sesame app edge](https://user-images.githubusercontent.com/43503404/76238363-c1b79e80-6255-11ea-9eaa-50efef756c7b.jpg)
![Screenshot_2020-03-09-22-30-27-957_ninja sesame app edge](https://user-images.githubusercontent.com/43503404/76238368-c3816200-6255-11ea-8b7d-cd0e6ec21d5a.jpg)

I am using this without any launcher same as what i did in the previous device.
Using the sesame search widget only.

**Comment by philwall2** [comment]
It looks like Sesame hasn't detected any WhatsApp contacts.

1. Go to Sesame Settings > Shortcuts > Contacts
2. Do you see all your contacts listed there?
3. If not, do you have the Contacts permission granted? If you do and don't see them, try turning ON "Show hidden contacts"

If you are seeing you Contacts but nothing for WhatsApp, that would mean your Contacts DB is not flagging contacts as WhatsApp specific. Sesame uses that data to make the shortcuts. Our code could have a bug, but it's been working for 1 year+ so it's likely a data issue. LMK how things turn out.

**Comment by dhawalmehta22** [comment]
Yes, all the contacts are visible in the contacts tab as attached below. And yes the show hidden contacts is enabled there
![Screenshot_2020-03-19-15-07-26-703_ninja sesame app edge](https://user-images.githubusercontent.com/43503404/77053227-08468f00-69f4-11ea-9682-04df1f646bed.jpg)

But still no contacts are visible there in the whatsapp tab.
Do i need to do anything in the phone contact settings for this. Like what you mentioned for contacts DB flag.

**Comment by philwall2** [comment]
Unfortunately there's not much you can do. Your device's Contacts DB isn't flagging contacts as associated with WhatsApp. That's weird but TBH the Contacts DB is often a mess.

Your best bet is to manually add the WhatsApp conversations you use most often using the "WhatsApp Chat    +" button. That should only take about 5 minutes.

**Comment by dhawalmehta22** [comment]
Yah, surely i can do that.
But if i just wanted all my frequent contacts i would have created there shortcut on the home screen itself.
Like its one of the best reasons one can head towards sesame app.
Thats okay i will do that.
But this thing actually restricted me to buy this app from the store.
Wolud be greatful if future builds can resolve this issue.
